src/seth/libexec/seth/seth-bundle-source

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+### seth-bundle-source -- fetch source and compile contract
+### Usage: seth bundle-source <address> [<options>]
+###
+### Requires an etherscan api key
+###
+
+# not really sure why this is a list
+# seems like we always just want the first element
+set -e
+SRC=$(seth source "$1" | jq '.[0]')
+
+SOLC_VERSION=$(echo "$SRC" | jq '.CompilerVersion')
+SOLC_VERSION=${SOLC_VERSION/'"'}
+SOLC_VERSION=${SOLC_VERSION/v}
+SOLC_VERSION=${SOLC_VERSION/+*}
+
+SOURCE=$(echo "$SRC" | jq '.SourceCode')
+OPT=$(echo "$SRC" | jq '.OptimizationUsed' | seth --show-json)
+NAME=$(echo "$SRC" | jq '.ContractName' | seth --show-json)
+if [[ "$OPT" -ne  0 ]]; then
+    RUNS=$(echo "$SRC" | jq '.Runs')
+    export SOLC_FLAGS="--optimize optimize-runs="$RUNS""
+fi
+    
+srcfile=$(TMPDIR=. mktemp "$NAME".XXXXX)
```

Comment:
right now this leaves a whole lot of temporary files as junk. I wasn't quite sure how one might remove them because `seth debug` needs both the sourcefile (which is made here and the filename is lost) and the combined-json (which is knows about). I'm wondering what the right way to clean it up is...

Reply:
Dumb question: can you create them in `/tmp`? If not, in a local subdir that you remove at the end of the script (also in case of error, so adding a `trap`)?